src/main/java/org/opensearch/knn/index/engine/lucene/LuceneMethodResolver.java

```diff
@@ -94,6 +99,32 @@ private void validateConfig(KNNMethodConfigContext knnMethodConfigContext, boole
         }
     }
 
+    private void validateEncoderDimension(KNNMethodContext knnMethodContext, KNNMethodConfigContext knnMethodConfigContext) {
+        String encoderName = getEncoderName(knnMethodContext);
+        if (encoderName == null || ENCODER_SQ.equals(encoderName) == false) {
```

Comment:
Nit : You can just call `equals` for comparison. Nullity check will be done in there.

```
if (!ENCODER_DQ.equals(encoderName)) {
   ...
}
```